nucleus/payara-modules/nucleus-microprofile/config-service/src/main/java/fish/payara/nucleus/microprofile/config/source/SecretsDirConfigSource.java

```diff
@@ -143,39 +153,49 @@ private void findFile() {
         }
     }
 
-    private String readFile(String name) {
+    private String readFile(Path file) throws IOException {
         String result = null;
-        if (Files.exists(secretsDir) && Files.isDirectory(secretsDir) && Files.isReadable(secretsDir)) {
-            try {
-                Path file = Paths.get(secretsDir.toString(), name);
-                if (Files.exists(file) && Files.isReadable(file)) {
-                    StringBuilder collector = new StringBuilder();
-                    for (String line : Files.readAllLines(file)) {
-                        collector.append(line);
-                    }
-                    result = collector.toString();
-                }
-            } catch (IOException ex) {
-                Logger.getLogger(SecretsDirConfigSource.class.getName()).log(Level.SEVERE, null, ex);
+        if (Files.exists(file) && Files.isRegularFile(file) && Files.isReadable(file)) {
+            StringBuilder collector = new StringBuilder();
+            for (String line : Files.readAllLines(file)) {
+                collector.append(line);
             }
+            result = collector.toString();
```

Comment:
This should use `Files.readAllBytes` to create a new `String`

Reply:
OK will look into it. :smiley_cat:

Reply:
This has been taken care of.